flink/src/main/java/org/apache/iceberg/actions/RewriteDataFilesAction.java

```
@@ -0,0 +1,91 @@
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.iceberg.CombinedScanTask;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.flink.source.RowDataRewriter;
+import org.apache.iceberg.util.StructLikeWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RewriteDataFilesAction extends
+    RewriteDataFilesActionBase<RewriteDataFilesAction, RewriteDataFilesActionResult> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RewriteDataFilesAction.class);
+
+  private ExecutionEnvironment env = null;
+  private boolean caseSensitive;
+
+  public RewriteDataFilesAction(ExecutionEnvironment env, Table table) {
+    super(table);
+    this.env = env;
+    caseSensitive = Boolean.parseBoolean(table.properties().getOrDefault("case-sensitive", "false"));
+  }
+
+
+  @Override
+  public RewriteDataFilesActionResult execute() {
+    Map<StructLikeWrapper, Collection<FileScanTask>> filteredGroupedTasks = getFilteredGroupedTasks();
+
+    // Nothing to rewrite if there's only one DataFile in each partition.
+    if (filteredGroupedTasks.isEmpty()) {
+      return RewriteDataFilesActionResult.empty();
+    }
+
+    List<DataFile> currentDataFiles = getCurrentDataFiles(filteredGroupedTasks);
+
+    List<CombinedScanTask> combinedScanTasks = getCombinedScanTasks(filteredGroupedTasks);
+
+    DataSet<CombinedScanTask> dataSet = env.fromCollection(combinedScanTasks);
+    env.setParallelism(combinedScanTasks.size());
+
+    RowDataRewriter rowDataRewriter =
+        new RowDataRewriter(getTable(), getTable().spec(), caseSensitive, getTable().io(), getEncryptionManager());
+
+    List<DataFile> addedDataFiles = null;
+    try {
+      addedDataFiles = rowDataRewriter.rewriteDataForTasks(dataSet);
+    } catch (Exception e) {
+      LOG.error("rewrite data files error", e);
```

I think here we'd better to throw the exception to upper layer ?  How about throwing the `UncheckedIOException(e)` ?

---

This exception comes from the DataSet#collect method. I looked at this method again. It throws a RuntimeException. I think we also throw a RuntimeException here. What do you think?

---

Both `RuntimeException`  and `UncheckedIOException` works fine for me. Actually,  `UncheckedIOException` is also a `RuntimeException`.